### PR TITLE
fix memory issue in this test

### PR DESCRIPTION
**Pull request by jedwards4b**
Can you try this before I merge to develop and make sure that it works for you.

### CODE REVIEW

**Comment by jedwards4b** [comment]
I've added a cleanup of write_multi_nc functions which removes the unused maxiobufersize argument and merged to develop for testing.

**Comment by edhartnett** [comment]
Sorry, I didn't see your original request to try this until now.

I just tried it on develop branch and it looks good! Test is now working which failed before. Quick work, thanks. You came up with a fix in less time than some of us spent just arguing about process! ;-)

I am running my local jenkins suite on the develop branch. I'll let you know if anything turns up.

**Comment by dqwu** [comment]
@jedwards4b 
I checked out latest master branch and tested it on ANL workstation manually. 

As expected, test_darray_1d failed both Address Sanitizer check (found heap-buffer-overflow) and LeakSanitizer check (detected memory leaks).

Then I locally merged test_darray_1d_fix to the master branch and test_darray_1d passed both Address Sanitizer check and LeakSanitizer check.

I think you can go ahead to merge this PR to the develop branch for testing.

File: src/clib/pio_darray_int.c [hunk 1]
```diff
@@ -592,6 +590,7 @@ int pio_write_darray_multi_nc_serial(file_desc_t *file, int nvars, const int *vi
                         if ((mpierr = MPI_Recv(tmp_count, rregions * fndims, MPI_OFFSET, rtask, rtask + 3 * ios->num_iotasks,
                                                ios->io_comm, &status)))
                             return check_mpi(file, mpierr, __FILE__, __LINE__);
+			printf("%s %d %d %d %ld\n",__FILE__,__LINE__,nvars,rlen, iobuf);
```
Comment (edhartnett) [inline src/clib/pio_darray_int.c]: Did you mean to take out this printf?

File: tests/cunit/test_darray_1d.c [hunk 2]
```diff
@@ -574,10 +574,6 @@ int main(int argc, char **argv)
             /* Run tests for each data type. */
             for (int t = 0; t < NUM_TYPES_TO_TEST; t++)
             {
-                /* This combination is still broken. */
-                if (rearranger[r] == PIO_REARR_SUBSET && test_type[t] == PIO_DOUBLE)
-                    continue;
-
```
Comment (edhartnett) [inline tests/cunit/test_darray_1d.c]: Nice work!

**Comment by jedwards4b** [comment]
removed debug print statement and updated develop

**Pull request merged**